recursion error
```
E   RecursionError: maximum recursion depth exceeded while calling a Python object
!!! Recursion detected (same locals & position)
__ ERROR collecting tests/providers/cncf/kubernetes/utils/test_pod_manager.py __
tests/providers/cncf/kubernetes/utils/test_pod_manager.py:31: in <module>
    from airflow.providers.cncf.kubernetes.utils.pod_manager import PodManager, PodPhase, container_is_running
airflow/providers/cncf/kubernetes/utils/pod_manager.py:40: in <module>
    from airflow.kubernetes.kube_client import get_kube_client
airflow/kubernetes/kube_client.py:88: in <module>
    in_cluster: bool = conf.getboolean("kubernetes_executor", "in_cluster"),
airflow/configuration.py:742: in getboolean
    val = str(self.get(section, key, **kwargs)).lower().strip()
airflow/configuration.py:658: in get
    return self.get(old_section, key, **kwargs)
airflow/configuration.py:658: in get
    return self.get(old_section, key, **kwargs)
E   RecursionError: maximum recursion depth exceeded
!!! Recursion detected (same locals & position)
_ ERROR collecting tests/providers/google/cloud/operators/test_kubernetes_engine.py _
tests/providers/google/cloud/operators/test_kubernetes_engine.py:31: in <module>
    from airflow.providers.cncf.kubernetes.operators.kubernetes_pod import KubernetesPodOperator
airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py:46: in <module>
    from airflow.providers.cncf.kubernetes.hooks.kubernetes import KubernetesHook
airflow/providers/cncf/kubernetes/hooks/kubernetes.py:29: in <module>
    from airflow.kubernetes.kube_client import _disable_verify_ssl, _enable_tcp_keepalive
airflow/kubernetes/kube_client.py:88: in <module>
    in_cluster: bool = conf.getboolean("kubernetes_executor", "in_cluster"),
airflow/configuration.py:742: in getboolean
    val = str(self.get(section, key, **kwargs)).lower().strip()
airflow/configuration.py:658: in get
    return self.get(old_section, key, **kwargs)
airflow/configuration.py:658: in get
    return self.get(old_section, key, **kwargs)
E   RecursionError: maximum recursion depth exceeded while calling a Python object
```

--- a/airflow/configuration.py
+++ b/airflow/configuration.py
@@ -233,7 +233,7 @@ class AirflowConfigParser(ConfigParser):
     }
 
     # A mapping of new section -> (old section, since_version).
-    deprecated_sections: dict[str, tuple[str, str]] = {"kubernetes_executor": ("kubernetes", "2.4.2")}
+    deprecated_sections: dict[str, tuple[str, str]] = {"kubernetes_executor": ("kubernetes", "2.5.0")}
 
     # Now build the inverse so we can go from old_section/old_key to new_section/new_key
     # if someone tries to retrieve it based on old_section/old_key

--- a/airflow/configuration.py
+++ b/airflow/configuration.py
@@ -229,33 +229,6 @@ class AirflowConfigParser(ConfigParser):
         ("database", "sql_alchemy_connect_args"): ("core", "sql_alchemy_connect_args", "2.3.0"),
         ("database", "load_default_connections"): ("core", "load_default_connections", "2.3.0"),
         ("database", "max_db_retries"): ("core", "max_db_retries", "2.3.0"),
-        **{
-            ("kubernetes_executor", x): ("kubernetes", x, "2.4.2")
-            for x in (
-                "pod_template_file",
-                "worker_container_repository",
-                "worker_container_tag",
-                "namespace",
-                "delete_worker_pods",
-                "delete_worker_pods_on_failure",
-                "worker_pods_creation_batch_size",
-                "multi_namespace_mode",
-                "in_cluster",
-                "cluster_context",
-                "config_file",
-                "kube_client_request_args",
-                "delete_option_kwargs",
-                "enable_tcp_keepalive",
-                "tcp_keep_idle",
-                "tcp_keep_intvl",
-                "tcp_keep_cnt",
-                "verify_ssl",
-                "worker_pods_pending_timeout",
-                "worker_pods_pending_timeout_check_interval",
-                "worker_pods_queued_check_interval",
-                "worker_pods_pending_timeout_batch_size",
-            )
-        },
         ("scheduler", "parsing_cleanup_interval"): ("scheduler", "deactivate_stale_dags_interval", "2.5.0"),
     }
 
@@ -593,28 +566,35 @@ class AirflowConfigParser(ConfigParser):
         deprecated_key: str | None
 
         # For when we rename whole sections
-        if section in self.deprecated_sections:
+        if section in self.inversed_deprecated_sections:
+            deprecated_section, deprecated_key = (section, key)
+            section = self.inversed_deprecated_sections[section]
             warnings.warn(
-                f"The config section [{section}] has been renamed to "
-                f"[{self.deprecated_sections[section][0]}]. Please update your config.",
+                f"The config section [{deprecated_section}] has been renamed to "
+                f"[{section}]. Please update your `conf.get*` call to use the new name",
                 FutureWarning,
                 stacklevel=2,
             )
             # Don't warn about individual rename if the whole section is renamed
             issue_warning = False
-            deprecated_section, deprecated_key = (section, key)
-            section = self.deprecated_sections[section][0]
         elif (section, key) in self.inversed_deprecated_options:
             # Handle using deprecated section/key instead of the new section/key
-            deprecated_section, deprecated_key = self.inversed_deprecated_options[(section, key)]
+            new_section, new_key = self.inversed_deprecated_options[(section, key)]
             if issue_warning:
                 warnings.warn(
-                    f"section/key [{deprecated_section}/{deprecated_key}] has been deprecated, "
-                    f"you should use [{section}/{key}] instead.",
-                    DeprecationWarning,
+                    f"section/key [{section}/{key}] has been deprecated, you should use"
+                    f"[{new_section}/{new_key}] instead. Please update your `conf.get*` call to use the "
+                    "new name",
+                    FutureWarning,
                     stacklevel=2,
                 )
                 issue_warning = False
+            deprecated_section, deprecated_key = section, key
+            section, key = (new_section, new_key)
+        elif section in self.deprecated_sections:
+            # When accessing the new section name, make sure we check under the old config name
+            deprecated_key = key
+            deprecated_section = self.deprecated_sections[section][0]
         else:
             deprecated_section, deprecated_key, _ = self.deprecated_options.get(
                 (section, key), (None, None, None)
@@ -652,14 +632,9 @@ class AirflowConfigParser(ConfigParser):
         if self.airflow_defaults.has_option(section, key) or "fallback" in kwargs:
             return expand_env_var(self.airflow_defaults.get(section, key, **kwargs))
 
-        # Finally attempt: look for the old section name we have renamed en-mass
-        elif section in self.inversed_deprecated_sections:
-            old_section = self.inversed_deprecated_sections[section]
-            return self.get(old_section, key, **kwargs)
-        else:
-            log.warning("section/key [%s/%s] not found in config", section, key)
+        log.warning("section/key [%s/%s] not found in config", section, key)
 
-            raise AirflowConfigException(f"section/key [{section}/{key}] not found in config")
+        raise AirflowConfigException(f"section/key [{section}/{key}] not found in config")
 
     def _get_option_from_secrets(
         self,

--- a/tests/core/test_configuration.py
+++ b/tests/core/test_configuration.py
@@ -54,6 +54,15 @@ from tests.utils.test_config import (
 HOME_DIR = os.path.expanduser("~")
 
 
+@pytest.fixture(scope="module")
+def restore_env():
+    current = os.environ.copy()
+
+    yield
+
+    os.environ = current
+
+
 @unittest.mock.patch.dict(
     "os.environ",
     {
@@ -63,6 +72,7 @@ HOME_DIR = os.path.expanduser("~")
         "AIRFLOW__TESTCMDENV__NOTACOMMAND_CMD": 'echo -n "NOT OK"',
     },
 )
+@pytest.mark.usefixtures("restore_env")
 class TestConf:
     def test_airflow_home_default(self):
         with unittest.mock.patch.dict("os.environ"):

--- a/tests/core/test_configuration.py
+++ b/tests/core/test_configuration.py
@@ -871,6 +871,10 @@ class TestDeprecatedConf:
                 with mock.patch.dict("os.environ", AIRFLOW__CORE__LOGGING_LEVEL="VALUE"):
                     assert conf.get("logging", "logging_level") == "VALUE"
 
+            with pytest.warns(FutureWarning, match="Please update your `conf.get"):
+                with mock.patch.dict("os.environ", AIRFLOW__CORE__LOGGING_LEVEL="VALUE"):
+                    assert conf.get("core", "logging_level") == "VALUE"
+
             with pytest.warns(DeprecationWarning), conf_vars({("core", "logging_level"): "VALUE"}):
                 assert conf.get("logging", "logging_level") == "VALUE"
 
@@ -1046,10 +1050,7 @@ sql_alchemy_conn=sqlite://test
             # Guarantee we have a deprecated setting, so we test the deprecation
             # lookup even if we remove this explicit fallback
             test_conf.deprecated_sections = {
-                "old_section": ("new_section", "2.1"),
-            }
-            test_conf.deprecated_options = {
-                ("old_section", "val"): ("new_section", "val", "2.1"),
+                "new_section": ("old_section", "2.1"),
             }
             test_conf.read_dict(conf_dict)
             test_conf.validate()
@@ -1061,14 +1062,15 @@ sql_alchemy_conn=sqlite://test
         test_conf = make_config()
         with pytest.warns(
             DeprecationWarning,
-            match=r"section/key \[old_section/val\] has been deprecated, you should use "
-            r"\[new_section/val\] instead",
+            match=r"\[old_section\] has been moved to the val option in \[new_section\].*update your config",
         ):
+            # Test when you've _set_ the old value that we warn you need to update your config
             assert test_conf.get("new_section", "val") == expected
         with pytest.warns(
             FutureWarning,
-            match=r"The config section \[old_section\] has been renamed to \[new_section\]",
+            match=r"\[old_section\] has been renamed to \[new_section\].*update your `conf.get",
         ):
+            # Test when you read using the old section you get told to change your `conf.get` call
             assert test_conf.get("old_section", "val") == expected
 
     def test_deprecated_funcs(self):

